bench_test.go

```diff
@@ -0,0 +1,264 @@
+package yarpc_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/yarpc/yarpc-go"
+	"github.com/yarpc/yarpc-go/encoding/raw"
+	"github.com/yarpc/yarpc-go/transport"
+	yhttp "github.com/yarpc/yarpc-go/transport/http"
+	ytchannel "github.com/yarpc/yarpc-go/transport/tchannel"
+
+	"github.com/stretchr/testify/require"
+	"github.com/uber/tchannel-go"
+	traw "github.com/uber/tchannel-go/raw"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+func yarpcEcho(reqMeta yarpc.ReqMeta, body []byte) ([]byte, yarpc.ResMeta, error) {
+	return body, yarpc.NewResMeta(reqMeta.Context()).Headers(reqMeta.Headers()), nil
+}
+
+var httpEcho http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	hs := w.Header()
+	for k, vs := range r.Header {
+		for _, v := range vs {
+			hs.Set(k, v)
+		}
+	}
+
+	_, err := io.Copy(w, r.Body)
+	if err != nil {
+		panic(fmt.Sprintf("HTTP handler failed: %v", err))
```

Comment:
rather than panic, you could take a `testing.TB` and have the test fail,
```go
func httpEcho(t testing.TB) http.HandlerFunc {
  return func(w http.ResponseWriter, r *http.Request) {
    [...]
  }
}
```

This also avoids the `var httpEcho` declaration which always is a little odd.